superset/assets/javascripts/explorev2/components/FilterField.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { PropTypes } from 'react';
+import { Button, Row, Col } from 'react-bootstrap';
+import Filter from './Filter';
+
+const propTypes = {
+  prefix: PropTypes.string,
+  choices: PropTypes.array,
+  onChange: PropTypes.func,
+  value: PropTypes.array,
+  datasource: PropTypes.object,
+};
+
+const defaultProps = {
+  prefix: 'flt',
+  choices: [],
+  onChange: () => {},
+  value: [],
+};
+
+export default class FilterField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.opChoices = props.prefix === 'flt' ?
+      ['in', 'not in'] : ['==', '!=', '>', '<', '>=', '<='];
+  }
+  addFilter() {
+    const newFilters = this.props.value;
+    newFilters.push({
+      prefix: this.props.prefix,
+      col: null,
+      op: 'in',
+      value: this.props.datasource.filter_select ? [] : '',
+    });
+    this.props.onChange(newFilters);
+  }
+  changeFilter(index, field, value) {
+    const newFilters = this.props.value;
+    const modifiedFilter = Object.assign({}, newFilters[index]);
+    modifiedFilter[field] = value;
+    newFilters.splice(index, 1, modifiedFilter);
+    this.props.onChange(newFilters);
+  }
+  removeFilter(index) {
+    const newFilters = this.props.value;
```

Comment:
That would be mutating it, you need to `Object.assign({}, this.props.value)` first, or more simply `this.props.value.filter((f, i) => i !== 1)`